# Prieskumná analýza datasetu č.17

Autori: Andrea Dická a Ján Nemčík
Cvičenie: Piatok, 8:00


In [10]:
%matplotlib inline
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sb


In [2]:
data = pd.read_csv("17/test.csv")
data.head()

Unnamed: 0  age sex query on thyroxine on antithyroid medication  \
0           0   24   M                  f                         f   
1           1   69   F                  f                         f   
2           2   37   M                  f                         f   
3           3   46   M                  f                         f   
4           4   29   F                  f                         f   

  thyroid surgery I131 treatment tumor hypopituitary   TSH  \
0               f              f     f             f  3.30   
1               f              f     f             f  1.10   
2               f              f     f             f  2.60   
3               f              f     f             f  0.25   
4               f              f     f             f  1.60   

                         ...                               marital-status  \
0                        ...                           Married-civ-spouse   
1                        ...                                     Divorced   
2                        ...                           Married-civ-spouse   
3                        ...                           Married-civ-spouse   
4                        ...                           Married-civ-spouse   

          occupation    relationship    race  capital-gain capital-loss  \
0     Prof-specialty         Husband   White             0            0   
1      Other-service   Not_in_family   White             0            0   
2       Craft-repair         Husband   White             0         1672   
3   Transport-moving         Husband   Black             0            0   
4    Farming-fishing            Wife   White             0            0   

  hours-per-week  native-country        date_of_birth  \
0             40   United-States           1993/08/27   
1             32   United-States  1949-03-07 00 00 00   
2             48   United-States           1980-09-12   
3             40   United-States           15/07/1971   
4             60   United-States           1988-08-17   

                                        medical_info  
0  {'goitre':'f','TSH measured':'t','pregnant':'f...  
1  {'goitre':'f','TSH measured':'t','pregnant':'f...  
2  {'goitre':'f','TSH measured':'t','pregnant':'f...  
3  {'goitre':'f','TSH measured':'t','pregnant':'f...  
4  {'goitre':'f','TSH measured':'t','pregnant':'f...  

[5 rows x 35 columns]

Hneď sme si všimli, že prvý stĺpec je nepomenovaný a vyjadruje iba číslo riadku, takže sme sa rozhodli, že je nepotrebný.

In [3]:
data = pd.read_csv("17/test.csv", sep=',', header=None, index_col=0)
data.head()

1    2                   3                          4   \
0                                                               
NaN   age  sex  query on thyroxine  on antithyroid medication   
 0.0   24    M                   f                          f   
 1.0   69    F                   f                          f   
 2.0   37    M                   f                          f   
 3.0   46    M                   f                          f   

                   5               6      7              8     9   \
0                                                                   
NaN   thyroid surgery  I131 treatment  tumor  hypopituitary   TSH   
 0.0                f               f      f              f   3.3   
 1.0                f               f      f              f   1.1   
 2.0                f               f      f              f   2.6   
 3.0                f               f      f              f  0.25   

               10                        ...                          \
0                                        ...                           
NaN   T3 measured                        ...                           
 0.0            t                        ...                           
 1.0            t                        ...                           
 2.0            t                        ...                           
 3.0            t                        ...                           

                       25                 26              27      28  \
0                                                                      
NaN        marital-status         occupation    relationship    race   
 0.0   Married-civ-spouse     Prof-specialty         Husband   White   
 1.0             Divorced      Other-service   Not_in_family   White   
 2.0   Married-civ-spouse       Craft-repair         Husband   White   
 3.0   Married-civ-spouse   Transport-moving         Husband   Black   

                29            30              31              32  \
0                                                                  
NaN   capital-gain  capital-loss  hours-per-week  native-country   
 0.0             0             0              40   United-States   
 1.0             0             0              32   United-States   
 2.0             0          1672              48   United-States   
 3.0             0             0              40   United-States   

                       33                                                 34  
0                                                                             
NaN         date_of_birth                                       medical_info  
 0.0           1993/08/27  {'goitre':'f','TSH measured':'t','pregnant':'f...  
 1.0  1949-03-07 00 00 00  {'goitre':'f','TSH measured':'t','pregnant':'f...  
 2.0           1980-09-12  {'goitre':'f','TSH measured':'t','pregnant':'f...  
 3.0           15/07/1971  {'goitre':'f','TSH measured':'t','pregnant':'f...  

[5 rows x 34 columns]

In [4]:
data.columns = ["{:02d}".format(i) for i in range(len(data.columns))]
data.head()

00   01                  02                         03  \
0                                                               
NaN   age  sex  query on thyroxine  on antithyroid medication   
 0.0   24    M                   f                          f   
 1.0   69    F                   f                          f   
 2.0   37    M                   f                          f   
 3.0   46    M                   f                          f   

                   04              05     06             07    08  \
0                                                                   
NaN   thyroid surgery  I131 treatment  tumor  hypopituitary   TSH   
 0.0                f               f      f              f   3.3   
 1.0                f               f      f              f   1.1   
 2.0                f               f      f              f   2.6   
 3.0                f               f      f              f  0.25   

               09                        ...                          \
0                                        ...                           
NaN   T3 measured                        ...                           
 0.0            t                        ...                           
 1.0            t                        ...                           
 2.0            t                        ...                           
 3.0            t                        ...                           

                       24                 25              26      27  \
0                                                                      
NaN        marital-status         occupation    relationship    race   
 0.0   Married-civ-spouse     Prof-specialty         Husband   White   
 1.0             Divorced      Other-service   Not_in_family   White   
 2.0   Married-civ-spouse       Craft-repair         Husband   White   
 3.0   Married-civ-spouse   Transport-moving         Husband   Black   

                28            29              30              31  \
0                                                                  
NaN   capital-gain  capital-loss  hours-per-week  native-country   
 0.0             0             0              40   United-States   
 1.0             0             0              32   United-States   
 2.0             0          1672              48   United-States   
 3.0             0             0              40   United-States   

                       32                                                 33  
0                                                                             
NaN         date_of_birth                                       medical_info  
 0.0           1993/08/27  {'goitre':'f','TSH measured':'t','pregnant':'f...  
 1.0  1949-03-07 00 00 00  {'goitre':'f','TSH measured':'t','pregnant':'f...  
 2.0           1980-09-12  {'goitre':'f','TSH measured':'t','pregnant':'f...  
 3.0           15/07/1971  {'goitre':'f','TSH measured':'t','pregnant':'f...  

[5 rows x 34 columns]

In [5]:
data.count()

00    973
01    973
02    973
03    973
04    973
05    973
06    973
07    973
08    888
09    973
10    789
11    973
12    926
13    883
14    973
15    973
16    973
17    973
18    973
19    973
20    973
21    973
22    973
23    834
24    973
25    973
26    973
27    973
28    973
29    973
30    973
31    973
32    973
33    973
dtype: int64

Vidíme, že máme k dispozícii 973 záznamov. Väčšina stĺpcov obsahuje všetky dáta, iba v piatich stĺpcoch sú nejaké chýbajúce dáta.

In [6]:
data = data.drop(data.columns[2:8], axis = 1)
data.head()

00   01    08           09   10            11     12    13  \
0                                                                   
NaN   age  sex   TSH  T3 measured   T3  TT4 measured    TT4   T4U   
 0.0   24    M   3.3            t  1.2             t   70.0  0.87   
 1.0   69    F   1.1            t  1.9             t  135.0  0.96   
 2.0   37    M   2.6            t  2.3             t   68.0  0.94   
 3.0   46    M  0.25            t  2.0             t  112.0  1.01   

                14   15                        ...                          \
0                                              ...                           
NaN   FTI measured  FTI                        ...                           
 0.0             t   80                        ...                           
 1.0             t  141                        ...                           
 2.0             t   72                        ...                           
 3.0             t  111                        ...                           

                       24                 25              26      27  \
0                                                                      
NaN        marital-status         occupation    relationship    race   
 0.0   Married-civ-spouse     Prof-specialty         Husband   White   
 1.0             Divorced      Other-service   Not_in_family   White   
 2.0   Married-civ-spouse       Craft-repair         Husband   White   
 3.0   Married-civ-spouse   Transport-moving         Husband   Black   

                28            29              30              31  \
0                                                                  
NaN   capital-gain  capital-loss  hours-per-week  native-country   
 0.0             0             0              40   United-States   
 1.0             0             0              32   United-States   
 2.0             0          1672              48   United-States   
 3.0             0             0              40   United-States   

                       32                                                 33  
0                                                                             
NaN         date_of_birth                                       medical_info  
 0.0           1993/08/27  {'goitre':'f','TSH measured':'t','pregnant':'f...  
 1.0  1949-03-07 00 00 00  {'goitre':'f','TSH measured':'t','pregnant':'f...  
 2.0           1980-09-12  {'goitre':'f','TSH measured':'t','pregnant':'f...  
 3.0           15/07/1971  {'goitre':'f','TSH measured':'t','pregnant':'f...  

[5 rows x 28 columns]

In [7]:
data = data.drop(['16','17'],axis=1)
data.head()

00   01    08           09   10            11     12    13  \
0                                                                   
NaN   age  sex   TSH  T3 measured   T3  TT4 measured    TT4   T4U   
 0.0   24    M   3.3            t  1.2             t   70.0  0.87   
 1.0   69    F   1.1            t  1.9             t  135.0  0.96   
 2.0   37    M   2.6            t  2.3             t   68.0  0.94   
 3.0   46    M  0.25            t  2.0             t  112.0  1.01   

                14   15                        ...                          \
0                                              ...                           
NaN   FTI measured  FTI                        ...                           
 0.0             t   80                        ...                           
 1.0             t  141                        ...                           
 2.0             t   72                        ...                           
 3.0             t  111                        ...                           

                       24                 25              26      27  \
0                                                                      
NaN        marital-status         occupation    relationship    race   
 0.0   Married-civ-spouse     Prof-specialty         Husband   White   
 1.0             Divorced      Other-service   Not_in_family   White   
 2.0   Married-civ-spouse       Craft-repair         Husband   White   
 3.0   Married-civ-spouse   Transport-moving         Husband   Black   

                28            29              30              31  \
0                                                                  
NaN   capital-gain  capital-loss  hours-per-week  native-country   
 0.0             0             0              40   United-States   
 1.0             0             0              32   United-States   
 2.0             0          1672              48   United-States   
 3.0             0             0              40   United-States   

                       32                                                 33  
0                                                                             
NaN         date_of_birth                                       medical_info  
 0.0           1993/08/27  {'goitre':'f','TSH measured':'t','pregnant':'f...  
 1.0  1949-03-07 00 00 00  {'goitre':'f','TSH measured':'t','pregnant':'f...  
 2.0           1980-09-12  {'goitre':'f','TSH measured':'t','pregnant':'f...  
 3.0           15/07/1971  {'goitre':'f','TSH measured':'t','pregnant':'f...  

[5 rows x 26 columns]

In [23]:
data = data.rename(columns=data.iloc[0]).drop(data.index[0])
data.head()

nan age sex   TSH T3 measured   T3 TT4 measured    TT4   T4U FTI measured  \
0                                                                           
0.0  24   M   3.3           t  1.2            t   70.0  0.87            t   
1.0  69   F   1.1           t  1.9            t  135.0  0.96            t   
2.0  37   M   2.6           t  2.3            t   68.0  0.94            t   
3.0  46   M  0.25           t  2.0            t  112.0  1.01            t   
4.0  29   F   1.6           t  2.6            t   89.0  0.98            t   

nan  FTI                        ...                          \
0                               ...                           
0.0   80                        ...                           
1.0  141                        ...                           
2.0   72                        ...                           
3.0  111                        ...                           
4.0   91                        ...                           

nan       marital-status         occupation    relationship    race  \
0                                                                     
0.0   Married-civ-spouse     Prof-specialty         Husband   White   
1.0             Divorced      Other-service   Not_in_family   White   
2.0   Married-civ-spouse       Craft-repair         Husband   White   
3.0   Married-civ-spouse   Transport-moving         Husband   Black   
4.0   Married-civ-spouse    Farming-fishing            Wife   White   

nan capital-gain capital-loss hours-per-week  native-country  \
0                                                              
0.0            0            0             40   United-States   
1.0            0            0             32   United-States   
2.0            0         1672             48   United-States   
3.0            0            0             40   United-States   
4.0            0            0             60   United-States   

nan        date_of_birth                                       medical_info  
0                                                                            
0.0           1993/08/27  {'goitre':'f','TSH measured':'t','pregnant':'f...  
1.0  1949-03-07 00 00 00  {'goitre':'f','TSH measured':'t','pregnant':'f...  
2.0           1980-09-12  {'goitre':'f','TSH measured':'t','pregnant':'f...  
3.0           15/07/1971  {'goitre':'f','TSH measured':'t','pregnant':'f...  
4.0           1988-08-17  {'goitre':'f','TSH measured':'t','pregnant':'f...  

[5 rows x 26 columns]

In [24]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Float64Index: 972 entries, 0.0 to 971.0
Data columns (total 26 columns):
age                972 non-null object
sex                972 non-null object
TSH                887 non-null object
T3 measured        972 non-null object
T3                 788 non-null object
TT4 measured       972 non-null object
TT4                925 non-null object
T4U                882 non-null object
FTI measured       972 non-null object
FTI                972 non-null object
referral source    972 non-null object
class              972 non-null object
workclass          972 non-null object
fnlwgt             972 non-null object
education          972 non-null object
education-num      833 non-null object
marital-status     972 non-null object
occupation         972 non-null object
relationship       972 non-null object
race               972 non-null object
capital-gain       972 non-null object
capital-loss       972 non-null object
hours-per-week     972 non-null

In [ ]:
for row in col[:]
    if:
        